src/main/java/com/voicepin/flow/client/FlowClient.java

```diff
@@ -1,45 +1,176 @@
+        Call<GetVoiceprintResult> call = new GetVoiceprintCall(getVoiceprintRequest);
+        return caller.call(call);
+    }
+
+    /**
+     * Creates new voice model using streamed recording. Sets it for this Voiceprint. The model will be used as a
+     * reference in any subsequent verifications. If the Voiceprint has already been enrolled, its preexisting model
+     * would be overwritten.
+     * 
+     * @param enrollRequest request with voiceprint id and recording stream
+     * @return
+     * @throws FlowClientException
+     */
     public EnrollResult enroll(EnrollRequest enrollRequest) throws FlowClientException {
         Call<EnrollResult> call = new EnrollCall(enrollRequest);
         return caller.call(call);
     }
 
-    public VerifyResult verify(VerifyRequest verifyRequest) throws FlowClientException {
+    /**
+     * Starts verification process on given Voiceprint with incoming stream. Verification process starts immediately
+     * (i.e. it does not wait for the whole stream to be uploaded).
+     *
+     * @param verifyRequest
+     * @return result client for getting results
+     * @throws FlowClientException
+     */
+    public VerifyStreamClient verify(VerifyRequest verifyRequest) throws FlowClientException {
+
         VerifyInitRequest initReq = new VerifyInitRequest(verifyRequest.getVoiceprintId());
         Call<VerifyInitResult> initCall = new VerifyInitCall(initReq);
         VerifyInitResult initResult = caller.call(initCall);
 
-        VerifyStreamRequest streamReq = new VerifyStreamRequest(initResult.getSpeechPath(), verifyRequest.getSpeechStream());
-        Call<VerifyStreamResult> streamCall = new VerifyStreamCall(streamReq);
-        VerifyStreamResult streamResult = caller.call(streamCall);
+        return new VerifyStreamClient(caller, initResult, verifyRequest.getSpeechStream());
+    }
+
+    /**
+     * Creates builder for FlowClient.
+     * 
+     * @param baseUrl url to the service eg.
+     *            http://flow.voicepin.com/voicepin-ti-server/v1/
+     *
+     * @return
+     */
+    public static FlowClientBuilder newBuilder(String baseUrl) {
+        return new FlowClientBuilder(baseUrl);
+    }
+
+    public static final class FlowClientBuilder {
+
+        private final String baseUrl;
+
+        private String username;
+        private String password;
+
+        private CertificateStrategy connectionHelper = new TrustedCertificateStrategy();
+
+        private FlowClientBuilder(String baseUrl) {
+            this.baseUrl = baseUrl;
+        }
+
+        /**
+         * Allows to connect to the Voicepin Flow server using secure http connection with
+         * given credentials.
+         * 
+         * @param username
+         * @param password
+         *
+         * @return
+         */
+        public FlowClientBuilder withHttps(String username, String password) {
```

Comment:
Przecież HTTPS w standardzie nie wymaga autentykacji. To powinno być niezależne np. jako withBasicAuth(username, password).